packages/core/src/codecatalyst/auth.ts

```diff
@@ -118,6 +155,54 @@ export class CodeCatalystAuthenticationProvider {
         await this.secondaryAuth.restoreConnection()
     }
 
+    private async accessDeniedExceptionHandler() {
+        if (!this.isConnectionValid()) return
+
+        // Check if CodeWhisper and CodeCatalyst share the same connection
+        const isSharedConn =
+            this.secondaryAuth.activeConnection?.id == CodeWhispererAuth.instance.secondaryAuth.activeConnection?.id
+
+        if (isSharedConn) {
+            await codeWhispererClient.listAvailableCustomizations()
+        }
+
+        this.updateConnectionState(this.activeConnection!, { shared: isSharedConn })
```

Comment:
Why is an extra `shared` flag needed? Can we just calculate it every time, instead (could be abstracted in a `isSharedConn()` function if necessary?